Update docker and gitlab-runner

gurinderu opened the pull request:
-update docker to 17.05
-update gitlab-runner to 9.4.2

Commit:
Refactoring
- using gitlab/gitlab-runner:alpine as parent image
- dnd hack was changed to wrapdocker
- rename some envs
- update README.md

ryadav88 commented:
@tobilg can I ask you to review this PR?

tobilg commented:
I already chatted with @gurinderu last week. I'm not a big fan of using a non-supported (officially) Docker version. 

Reason behind that was if I understood correctly that they wanted to use newer Docker build features. 

Also, afaik, the GitLab package itself is not yet updated to the same version.

gurinderu commented:
@tobilg Hey again. 
I partly refactored your image. Could you review again?
- I changed parent image to gitlab-runner:alpine which already contains dumb-init and gitlab
- I changed some envs, because I think we should use same env names like for gitlab-runner
Don't worry about docker

gurinderu commented:
@tobilg yeap, we need to use new docker,because it has a multistage build.
And don't worry about docker, because it is a dind.

tobilg commented on register_and_run.sh:
```diff
@@ -134,7 +140,10 @@ _getTerminationSignal() {
 
 # Start the runner
-gitlab-runner run --working-directory=${RUNNER_WORK_DIR} --metrics-server $HOST:$PORT0
+gitlab-ci-multi-runner run --user=gitlab-runner --working-directory=${RUNNER_WORK_DIR} --metrics-server :$PORT0
```
Why was `$HOST` removed?

gurinderu replied:
@tobilg please don't merge it, because it is working wrong for a dind. Looks like gitlab or docker bug.  

P.S It was removed because runner can't start with specified host

tobilg commented on register_and_run.sh:
```diff
@@ -111,8 +110,15 @@ else
 echo "==> Launching the Docker daemon..."
-dind docker daemon --host=unix:///var/run/docker.sock --storage-driver=overlay $DOCKER_EXTRA_OPTS &
+
+dind dockerd --host=unix:///var/run/docker.sock --storage-driver=${DOCKER_VOLUME_DRIVER}&
```
Why was `$DOCKER_EXTRA_OPTS` removed?

tobilg commented on register_and_run.sh:
```diff
@@ -134,7 +140,10 @@ _getTerminationSignal() {
-gitlab-runner register
+gitlab-ci-multi-runner register \
+  --name ${MESOS_TASK_ID} \
+  --docker-privileged \
```
Are `--docker-privileged` and `--docker-volume-driver` real flags here? I have trouble finding them in the docs...

tobilg commented:
@ryadav88 I don't have a DC/OS cluster at hand at the moment, so I can't really test this. Also, I think the updated GitLab Runner version needs to match to the GitLab package version...

tobilg reviewed (commented):
Regarding `$HOST`, it worked for me perfectly before. Can you please elaborate regarding dind? Thanks!
